lib/core-integration/src/Test/Integration/Scenario/API/Transactions.hs

```diff
@@ -1025,26 +1043,338 @@ spec = do
+    --   1 | edge     | edge     | ascending  | 2 ascending  |
+    --   2 | edge     | edge + 1 | descending | 2 descending |
+    --   3 | edge     | edge - 1 | empty      | 1st one      |
+    --   4 | edge     | empty    | empty      | 2 descending |
+    --   5 | edge + 1 | edge + 1 | empty      | 2nd one      |
+    --   6 | edge + 1 | edge - 1 | empty      | none         |
+    --   7 | edge + 1 | empty    | ascending  | 2nd one      |
+    --   8 | edge + 1 | edge     | descending | 2nd one      |
+    --   9 | edge - 1 | edge - 1 | ascending  | 1st one      |
+    --  10 | edge - 1 | empty    | descending | 2 descending |
+    --  11 | edge - 1 | edge     | empty      | 2 descending |
+    --  12 | edge - 1 | edge + 1 | empty      | 2 descending |
+    --  13 | empty    | empty    | empty      | 2 descending |
+    --  14 | empty    | edge     | empty      | 2 descending |
+    --  15 | empty    | edge + 1 | ascending  | 2 ascending  |
+    --  16 | empty    | edge - 1 | descending | 1st one      |
+    -- +----+----------+----------+------------+--------------+
+    it "TRANS_LIST_02,03 - Can limit/order results with start, end and order" $ \ctx -> do
+        (_, w, [(t1, a1), (t2, a2)]) <- fixtureWalletManyTxs ctx [10,20]
+
+        let toParam = toQueryParam . utcIso8601ToText
+        let matrix =
+                [
+                -- 1
+                  ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam t2)
+                    <> "&order=ascending"
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a1
+                     , expectListItemFieldEqual 1 amount a2
+                     ]
+                  )
+                -- 2
+                , ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime 1 t2)
+                    <> "&order=descending"
+                  ,  [ expectListSizeEqual 2
+                     , expectListItemFieldEqual 0 amount a2
+                     , expectListItemFieldEqual 1 amount a1
+                     ]
+                  )
+                -- 3
+                , ( "?start="
+                    <> (toParam t1)
+                    <> "&end="
+                    <> (toParam $ addUTCTime (-1) t2)
+                  ,  [ expectListSizeEqual 1
+                     , expectListItemFieldEqual 0 amount a1
+                     ]
+                  )
+                -- 4
+                , ( "?start="
+                    <> (toParam t1)
+                  ,  [ expectListSizeEqual 2
+                      , expectListItemFieldEqual 0 amount a2
+                      , expectListItemFieldEqual 1 amount a1
+                      ]
+                    )
+                -- 5
+                , ( "?start="
+                    <> (toParam $ addUTCTime 1 t1)
```

Comment:
Scrolling down, it looks like there are a lot of these cases! I had a couple of ideas I wanted to share with you:

1. We could use named constants:

    ```hs
    onePicosecond :: NominalDiffTime
    onePicosecond = succ 0
    ```

Or

2. We could define a pair of functions to find the previous or next `UTCTime`:

    ```hs
    -- | For a given time 't0', get the closest representable time 't1' to 't0'
    --   for which 't0 < t1'.
    utcTimeSucc :: UTCTime -> UTCTime
    utcTimeSucc = addUTCTime (succ 0)

   -- | For a given time 't0', get the closest representable time 't1' to 't0'
    --   for which 't1 < t0'.
    utcTimePred :: UTCTime -> UTCTime
    utcTimePred = addUTCTime (pred 0)

    ```

What do you think?

I suspect using the functions might be clearer. (It's unfortunate that `UTCTime` cannot provide an `Enum` instance.)